js/search.js

```diff
@@ -0,0 +1,382 @@
+  });
+  addClickEventListenerToButtonsView();
+  firstActivateButtonView();
+}
+
+function firstActivateButtonView() {
+  let view = localStorage.getItem("view");
+  if (!view) {
+    view = "row";
+  }
+  let button = document.querySelectorAll(".view-button-" + view);
+  button = [].map.call(button, function(item) {
+    item.classList.add("view-button--active");
+  });
+  if (view === "row") {
+    let drinkContainers = document.getElementsByClassName("drink--wrapper");
+    drinkContainers = [].map.call(drinkContainers, function(item) {
+      item.classList.remove("drink--table");
+    });
+  } else {
+    let drinkContainers = document.getElementsByClassName("drink--wrapper");
+    drinkContainers = [].map.call(drinkContainers, function(item) {
+      item.classList.add("drink--table");
+    });
+  }
+}
+
+function bildButtonsView(elementPosition) {
+  const container = document.createElement("div");
+  container.classList.add("container--view");
+
+  const buttonTable = document.createElement("div");
+  buttonTable.classList.add("view-button");
+  buttonTable.classList.add("view-button-table");
+  buttonTable.innerHTML = "<i class='fas fa-th'></i>";
+
+  const buttonRow = document.createElement("div");
+  buttonRow.classList.add("view-button");
+  buttonRow.classList.add("view-button-row");
+  buttonRow.innerHTML = "<i class='fas fa-th-list'></i>";
+
+  container.appendChild(buttonTable);
+  container.appendChild(buttonRow);
+  elementPosition.appendChild(container);
+}
+
+function addClickEventListenerToButtonsView() {
+  let buttonRow = document.getElementsByClassName("view-button-row");
+  let buttonTable = document.getElementsByClassName("view-button-table");
+  buttonRow = [].map.call(buttonRow, function(item) {
+    item.addEventListener("click", function() {
+      localStorage.setItem("view", "row");
+      document.location.reload();
+    });
+  });
+  buttonTable = [].map.call(buttonTable, function(item) {
+    item.addEventListener("click", function() {
+      localStorage.setItem("view", "table");
+      document.location.reload();
+    });
+  });
+}
+
+function buildDrink(element, drinksPosistion) {
+  let position = drinksPosistion;
+  let drinkName = element.name.replace(/ /g, "-");
+  let wrapper = document.createElement("div");
+  wrapper.classList.add("drink--wrapper");
+  let container = document.createElement("div");
+  container.classList.add("drink");
+  let imgContainer = document.createElement("div");
+  imgContainer.classList.add("drink-img");
+  let link = document.createElement("a");
+  let containerInfo = document.createElement("div");
+  containerInfo.classList.add("drink-info");
+  let img = document.createElement("img");
+  img.src = element.image;
+  link.title = element.name;
+  link.href = "/pages/drink.html?drinks_" + element.id + "_" + drinkName;
+  link.appendChild(img);
+  imgContainer.appendChild(link);
+  container.appendChild(imgContainer);
+  let containerName = document.createElement("a");
```

Comment:
just use const)
and try to understand it fully
when you write
a = 1;
if nowhere in code you don't write a = ... again - it's const
if a is an object and you change one of it's properties - it should be const
because changing property doesn't mean you reassign the variable
same with arrays
push, splice and so on don't change the whole variable and this array also should be const